src/main/java/de/tum/in/www1/artemis/service/iris/IrisSessionService.java

```diff
@@ -58,6 +58,9 @@ public void checkIsIrisActivated(IrisSession session) {
      * @return The created session
      */
     public IrisSession createChatSessionForProgrammingExercise(ProgrammingExercise exercise, User user) {
+        if (exercise.isExamExercise()) {
+            throw new BadRequestException("Iris is not supported for exam exercises");
```

Comment:
We (@bassner and I) had something similar in one of my PRs and decided this should be a 409 (Conflict) as this is not a client error.

Reply:
Changed it 👍🏻